hoomd/md/update.py

```diff
@@ -221,100 +224,140 @@ class mueller_plathe_flow(_updater):
-        min_slab (int): Id < n_slabs where the min velocity component is search for. If set < 0 the value is set to its default 0.
+        filter (`hoomd.filter.ParticleFilter`): Subset of particles on which to
+            apply this updater.
 
-    .. attention::
-        * This updater has to be always applied every timestep.
-        * This updater works currently only with orthorhombic boxes.
+        flow_target (`hoomd.variant.Variant`): Integrated target flow. The unit
+            is in the natural units of the simulation: [flow_target] = [timesteps] x :math:`\mathcal{M}` x :math:`\frac{\mathcal{D}}{\tau}`.
+            The unit of [timesteps] is your discretization dt x :math:`\mathcal{\tau}`.
 
-    .. note:
-        If you set this updater with unrealistic values, it algorithm might not terminate,
-        because your desired flow target can not be achieved.
+        slab_direction (``X``, ``Y``, or ``Z``): Direction perpendicular to the slabs.
 
-    .. versionadded:: v2.1
+        flow_direction (``X``, ``Y``, or ``Z``): Direction of the flow.
 
-    Examples::
+        n_slabs (int): Number of slabs. You want as many as possible for small
+            disturbed volume, where the unphysical swapping is done. But each
+            slab has to contain a sufficient number of particle.
 
-        #const integrated flow with 0.1 slope for max 1e8 timesteps
-        const_flow = hoomd.variant.linear_interp( [(0,0),(1e8,0.1*1e8)] )
-        #velocity gradient in z direction and shear flow in x direction.
-        update.mueller_plathe_flow(all,const_flow,md.update.mueller_plathe_flow.Z,md.update.mueller_plathe_flow.X,100)
+        max_slab (int): Id < n_slabs where the max velocity component is search
+            for. If set < 0 the value is set to its default n_slabs/2.
 
-    """
-    def __init__(self, group,flow_target,slab_direction,flow_direction,n_slabs,max_slab=-1,min_slab=-1):
-        period=1 # This updater has to be applied every timestep
-        assert (n_slabs > 0 ),"Invalid negative number of slabs."
-        if min_slab < 0:
-            min_slab = 0
-        if max_slab < 0:
-            max_slab = n_slabs/2
-
-        #Cast input to int to avoid mismatch of types in calling the constructor
-        n_slabs = int(n_slabs)
-        min_slab = int(min_slab)
-        max_slab = int(max_slab)
-
-        assert (max_slab>-1 and max_slab < n_slabs),"Invalid max_slab in [0,"+str(n_slabs)+")."
-        assert (min_slab>-1 and min_slab < n_slabs),"Invalid min_slab in [0,"+str(n_slabs)+")."
-        assert (min_slab != max_slab),"Invalid min/max slabs. Both have the same value."
+        min_slab (int): Id < n_slabs where the min velocity component is search
+            for. If set < 0 the value is set to its default 0.
 
-        # initialize the base class
-        _updater.__init__(self);
+    Attention:
+        * This updater has to be always applied every timestep.
```

Comment:
@b-butler Can we force an Updater to have a certain trigger?

Reply:
Yes, through proper validation.

Reply:
Given Michael Howard's comments below, this may not be necessary. With a different implementation, the period does not need to be 1.

Reply:
Since you changed the code to always set the trigger to a Periodic(1), update the attention block to state that this is the trigger used.